proofs/proof_global.mli

```diff
@@ -13,18 +13,16 @@
      environment. *)
 
 type t
-type stack
 
-val get_current_pstate : stack -> t
-
-val get_current_proof_name : t -> Names.Id.t
-val get_current_persistence : t -> Decl_kinds.goal_kind
-val get_all_proof_names : stack -> Names.Id.t list
+(* Should be moved into a proper view *)
```

Comment:
Maybe we should expose the record `type t = private { ... }`. We already have getters for every field except endline_tactic, and it may be cleaner to access that through a getter or projection than passing it as argument to with_proof (which causes us to have to ignore the argument in most calls).

Reply:
Something like this is done in a later pr, which specifically focuses on moving and cleaning up fields, including some duplication. If you don't mind I'd like to address this comment on the upcoming PR.

Reply:
OK

Reply:
Merci